features/dd-sdk-android-rum/src/test/kotlin/com/datadog/android/rum/internal/domain/scope/RumSessionScopeTest.kt

```diff
@@ -1479,107 +1476,89 @@ internal class RumSessionScopeTest {
         )
 
         // Then
-        val context = checkNotNull(result).getRumContext()
-
-        assertThat(result).isSameAs(testedScope)
-        assertThat(context.sessionId).isNotEqualTo(RumContext.NULL_UUID)
-        assertThat(context.sessionState).isEqualTo(RumSessionScope.State.TRACKED)
-        assertThat(context.sessionStartReason).isEqualTo(RumSessionScope.StartReason.USER_APP_LAUNCH)
+        verify(mockRumSessionScopeStartupManager).onAppStartEvent(event = eq(event))
 
-        verify(mockRumAppStartupTelemetryReporter).reportTTID(eq(info), eq(0))
-        verifyNoMoreInteractions(mockRumAppStartupTelemetryReporter)
+        verifyNoMoreInteractions(mockRumSessionScopeStartupManager)
     }
 
     @ParameterizedTest
-    @MethodSource("testScenariosPairs")
-    fun `M call reportTTID twice { AppLaunchTTIDEvent }`(
-        scenario1: RumStartupScenario,
-        scenario2: RumStartupScenario,
+    @MethodSource("testScenarios")
+    fun `M call onTTIDEvent W handleEvent { AppLaunchTTIDEvent }`(
+        scenario: RumStartupScenario,
         forge: Forge
     ) {
         // Given
-        val info1 = RumTTIDInfo(
-            scenario = scenario1,
-            durationNs = forge.aLong(min = 0, max = 10000)
-        )
-
-        val info2 = RumTTIDInfo(
-            scenario = scenario2,
+        val info = RumTTIDInfo(
+            scenario = scenario,
             durationNs = forge.aLong(min = 0, max = 10000)
         )
 
-        val event1 = RumRawEvent.AppStartTTIDEvent(
-            info = info1
-        )
+        val event = RumRawEvent.AppStartTTIDEvent(info = info)
 
-        val event2 = RumRawEvent.AppStartTTIDEvent(
-            info = info2
-        )
-
-        // When
         testedScope.handleEvent(
-            event = event1,
+            event = fakeInitialViewEvent,
             datadogContext = fakeDatadogContext,
             writeScope = mockEventWriteScope,
             writer = mockWriter
         )
-        testedScope.handleEvent(
-            event = event2,
+
+        // When
+        val result = testedScope.handleEvent(
+            event = event,
             datadogContext = fakeDatadogContext,
             writeScope = mockEventWriteScope,
             writer = mockWriter
         )
 
+        val rumContext = checkNotNull(result).getRumContext()
+
         // Then
-        inOrder(mockRumAppStartupTelemetryReporter) {
-            verify(mockRumAppStartupTelemetryReporter).reportTTID(eq(info1), eq(0))
-            verify(mockRumAppStartupTelemetryReporter).reportTTID(eq(info2), eq(1))
-            verifyNoMoreInteractions()
-        }
+        verify(mockRumSessionScopeStartupManager).onTTIDEvent(
+            event = eq(event),
+            datadogContext = eq(fakeDatadogContext),
+            writeScope = eq(mockEventWriteScope),
+            writer = eq(mockWriter),
+            rumContext = eq(rumContext),
+            customAttributes = eq(fakeParentAttributes)
+        )
```

Comment:
nit: if all arguments are compared using `eq`, it is not really needed to write it explicitly. Some other calls in this test class can be updated as well.

```suggestion
        verify(mockRumSessionScopeStartupManager).onTTIDEvent(
            event = event,
            datadogContext = fakeDatadogContext,
            writeScope = mockEventWriteScope,
            writer = mockWriter,
            rumContext = rumContext,
            customAttributes = fakeParentAttributes
        )
```